tracing-error/src/backtrace.rs

```diff
@@ -120,6 +120,38 @@ impl SpanTrace {
             }
         });
     }
+
+    /// Returns the status of this SpanTrace, indicating whether this backtrace request was unsupported, empty, or a span trace was actually captured.
+    pub fn status(&self) -> SpanTraceStatus {
+        if self.span == Span::none() {
+            SpanTraceStatus::Empty
+        } else {
```

Comment:
the reason this doesn't work (and your test fails) is that the current `PartialEq` impl for `Span`s returns `false` if _either_ spans have no metadata, and doesn't have a case to handle _both_ spans having no metadata:
https://github.com/tokio-rs/tracing/blob/77e589906df9702a11f7c7bdb6ea69a3437a320f/tracing/src/span.rs#L835-L844

i think the original rationale was that two instances of `Span::none` may not originate from the same callsite, and thus might not be "equal". i'm not sure if this is how we want span equality to work or not, though, so perhaps we should change this so that two instances of `Span::none` _do_ equal each other...

regardless, a more robust solution (and also probably faster, as it doesn't create a second `Span::none`) would be to change this to use [`Span::is_none`](https://github.com/tokio-rs/tracing/blob/77e589906df9702a11f7c7bdb6ea69a3437a320f/tracing/src/span.rs#L730):
```suggestion
        if self.span.is_none() {
            SpanTraceStatus::Empty
        } else {
```
that will work regardless of whether or not we decide to change the equality rules for `Span::none`...